sormas-backend/src/main/java/de/symeda/sormas/backend/importexport/ImportFacadeEjb.java

```diff
@@ -529,6 +523,34 @@ public void generateEnvironmentImportTemplateFile(List<FeatureConfigurationDto>
 		writeTemplate(Paths.get(getEnvironmentImportTemplateFilePath()), importColumns, true);
 	}
 
+	@Override
+	public void generateSelfReportImportTemplateFile(List<FeatureConfigurationDto> featureConfigurations) throws IOException, NoSuchFieldException {
+
+		createExportDirectoryIfNecessary();
+
+		char separator = configFacade.getCsvSeparator();
+		List<ImportColumn> importColumns = new ArrayList<>();
+
+		appendListOfFields(
+			importColumns,
+			SelfReportDto.class,
+			"",
+			separator,
+			featureConfigurations,
+			SelfReportDto.class.getDeclaredField("investigationStatus"),
```

Comment:
reflection is slow so here please use strings only, and use constants from the dto class